credsweeper/file_handler/content_provider.py

```diff
@@ -28,7 +28,23 @@ def file_path(self) -> str:
     def lines_to_targets(self, lines: List[str]) -> List[AnalysisTarget]:
         """Creates list of targets with multiline concatenation"""
         targets = []
+        previous_target: Optional[AnalysisTarget] = None
         for i, line in enumerate(lines):
+            if '\\' == line[-1:]:
+                new_line = line[:-1]
+                if previous_target:
+                    # append current string to previous target
+                    previous_target.line += new_line
+                else:
+                    previous_target = AnalysisTarget(new_line, i + 1, lines, self.file_path)
+            else:
+                if previous_target:
+                    previous_target.line += line
+                    targets.append(previous_target)
+                    previous_target = None
+            # Anyway add original line to analysis
             target = AnalysisTarget(line, i + 1, lines, self.file_path)
             targets.append(target)
```

Comment:
I'm worried about adding original line to `targets` at every iteration..
If there is some code block like below, then user will receive duplicate report every time.
I think it would be good to add only merged line(`previous_target` + `line`) and original line(if there is no `'\\'`).
How do you think?

**Code block**

``` python
aws_client_id = \
"AKIAIANNKJJHPERXR0LA"
# This is example key
```

**Report**
``` bash
rule: AWS Client ID / severity: high / line_data_list: [line: 'aws_client_id = "AKIAIANNKJJHPERXR0LA"  ' / line_num: 1 / path: xx / value: 'AKIAIANNKJJHPERXR0LA' / entropy_validation: False] / api_validation: NOT_AVAILABLE / ml_validation: VALIDATED_KEY
rule: AWS Client ID / severity: high / line_data_list: [line: '"AKIAIANNKJJHPERXR0LA"  ' / line_num: 2 / path: xx / value: 'AKIAIANNKJJHPERXR0LA' / entropy_validation: False] / api_validation: NOT_AVAILABLE / ml_validation: VALIDATED_KEY
```

Reply:
I did not replace target due we have limit for string ````MAX_LINE_LENGTH = 1500``` - if a string is oversize - it is skipping :( even without warning.
So, merged string may be too long and possibly credentials are skipped.
Thus it is not a good source code, but credentials are badly. 
My opinion is to keep possibly duplicates when merged line is too long + to show the warning.

Reply:
Hmm.. I thought it more.. and still I think it is not good to keeping duplicates.. (Other services that using CredSweeper will be affected.)

So I suggest like this..

1. Manage and gather previous lines with list.
2. If gathered line exceed `MAX_LINE_LENGTH = 1500`.
   - True: Append each gathered lines to `targets`.
   - Else: Append gathered line to `targets`.

It might be a solution for keeping long line and each lines.